plugins/dell/fu-plugin-dell.c

```diff
@@ -341,23 +340,22 @@ fu_plugin_dock_node (FuPlugin *plugin, GUsbDevice *device,
 		return FALSE;
 	}
 
-	guid_str = g_strdup ("00000000-0000-0000-0000-000000000000");
-	if (efi_guid_to_str (guid_raw, &guid_str) < 0) {
-		g_debug ("Failed to convert GUID.");
-		return FALSE;
-	}
-
-	dock_key = fu_plugin_get_dock_key (plugin, device,
-						  guid_str);
+	dock_key = fu_plugin_get_dock_key (plugin, device, component_guid);
 	if (fu_plugin_cache_lookup (plugin, dock_key) != NULL) {
 		g_debug ("%s is already registered.", dock_key);
 		return FALSE;
 	}
 
 	dev = fu_device_new ();
-	dock_id = g_strdup_printf ("DELL-%s" G_GUINT64_FORMAT, guid_str);
-	dock_name = g_strdup_printf ("Dell %s %s", dock_type,
-				     component_desc);
+	dock_id = g_strdup_printf ("DELL-%s" G_GUINT64_FORMAT, component_guid);
+	if (component_desc != NULL) {
+		dock_name = g_strdup_printf ("Dell %s %s", dock_type,
+					     component_desc);
+		fu_device_add_parent_guid (dev, DOCK_FLASH_GUID);
+	}
+	else
```

Comment:
odd -- same line as brace and also braces for the lower half pls.

Reply:
fixed